src/main/java/com/hierynomus/sshj/userauth/keyprovider/OpenSSHKeyFileUtil.java

```diff
@@ -54,16 +56,19 @@ public static ParsedPubKey initPubKey(Reader publicKey) throws IOException {
                 if (!keydata.isEmpty()) {
                     String[] parts = keydata.trim().split("\\s+");
                     if (parts.length >= 2) {
+                        byte[] decodedPublicKey = Base64Decoder.decode(parts[1]);
                         return new ParsedPubKey(
                                 KeyType.fromString(parts[0]),
-                                new Buffer.PlainBuffer(Base64.getDecoder().decode(parts[1])).readPublicKey()
+                                new Buffer.PlainBuffer(decodedPublicKey).readPublicKey()
                         );
                     } else {
                         throw new IOException("Got line with only one column");
                     }
                 }
             }
             throw new IOException("Public key file is blank");
+        } catch (Base64DecodeError err) {
+            throw new IOException("Corrupted public key: " + err.getMessage(), err);
```

Comment:
Recommend adjusting the wording to be a bit more specific regarding the problem:
```suggestion
            throw new IOException("Public Key decoding failed: " + err.getMessage(), err);
```

Reply:
https://github.com/hierynomus/sshj/pull/936/commits/4e6ed1609b738dd85b7cda06517f853f471281d1